src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.cs

```diff
@@ -184,19 +167,13 @@ private static SymbolKind GetKindAndUnwrapAlias(ref ISymbol symbol)
         }
 
         private static bool IsConstructedFromSelf(INamedTypeSymbol symbol)
-        {
-            return symbol.Equals(symbol.ConstructedFrom);
-        }
+            => symbol.Equals(symbol.ConstructedFrom);
 
         private static bool IsConstructedFromSelf(IMethodSymbol symbol)
-        {
-            return symbol.Equals(symbol.ConstructedFrom);
-        }
+            => symbol.Equals(symbol.ConstructedFrom);
 
         private static bool IsObjectType(ISymbol symbol)
-        {
-            return symbol.Kind == SymbolKind.NamedType && ((ITypeSymbol)symbol).SpecialType == SpecialType.System_Object;
-        }
+            => symbol is ITypeSymbol typeSymbol && typeSymbol.SpecialType == SpecialType.System_Object;
```

Comment:
📝 This should keep the `Kind` check before the interface cast check for performance